lib/Controller/ChatController.php

```diff
@@ -255,18 +255,19 @@ public function sendMessage(string $message, string $actorDisplayName = '', stri
 	 * @param int $setReadMarker Automatically set the last read marker when 1,
 	 *                           if your client does this itself via chat/{token}/read set to 0
 	 * @param int $includeLastKnown Include the $lastKnownMessageId in the messages when 1 (default 0)
+	 * @param int $noStatusUpdate When the user status should not be automatically set to online set to 1 (default 0)
 	 * @return DataResponse an array of chat messages, "404 Not found" if the
 	 *         room token was not valid or "304 Not modified" if there were no messages;
 	 *         each chat message is an array with
 	 *         fields 'id', 'token', 'actorType', 'actorId',
 	 *         'actorDisplayName', 'timestamp' (in seconds and UTC timezone) and
 	 *         'message'.
 	 */
-	public function receiveMessages(int $lookIntoFuture, int $limit = 100, int $lastKnownMessageId = 0, int $timeout = 30, int $setReadMarker = 1, int $includeLastKnown = 0): DataResponse {
+	public function receiveMessages(int $lookIntoFuture, int $limit = 100, int $lastKnownMessageId = 0, int $timeout = 30, int $setReadMarker = 1, int $includeLastKnown = 0, int $noStatusUpdate = 0): DataResponse {
 		$limit = min(200, $limit);
 		$timeout = min(30, $timeout);
 
-		if ($this->participant->getSessionId() !== '0') {
+		if ($noStatusUpdate === 0 && $this->participant->getSessionId() !== '0') {
```

Comment:
Checking the parameter here would prevent not only user status updates but also pings. I do not know if that is intended or not :-)

Reply:
it is